PWGCF/FemtoDream/FemtoDreamParticleHisto.h

```diff
@@ -78,15 +83,104 @@ class FemtoDreamParticleHisto
+        } else if constexpr (mParticleType == o2::aod::femtodreamparticle::ParticleType::kV0) {
+          /// V0 histograms
+          mHistogramRegistry->fill(HIST(o2::aod::femtodreamparticle::ParticleTypeName[mParticleType]) + HIST(mFolderSuffix[mFolderSuffixType]) + HIST("/hCPA"),
+                                   part.pt(), part.tempFitVar());
+        } else if constexpr (mParticleType == o2::aod::femtodreamparticle::ParticleType::kCascade) {
+          /// Cascade histograms
+        } else {
+          LOG(fatal) << "FemtoDreamParticleHisto: Histogramming for requested object not defined - quitting!";
+        }
+      }
+    }
+  }
+
+  /// Initialization of the QA histograms for MonteCarlo ()
+  /// \param registry HistogramRegistry
+  template <typename T>
+  void initMC(HistogramRegistry* registry, T& ptBins, T& DCAxyBins)
+  {
+    if (registry) {
+      mHistogramRegistry = registry;
+
+      framework::AxisSpec ptAxis = {ptBins, "#it{p}_{T} (GeV/#it{c})"};
+      framework::AxisSpec DCAxyAxis = {DCAxyBins, "DCA_{xy} (cm)"};
+
+      /// The folder names are defined by the type of the object and the suffix (if applicable)
+      std::string folderName = static_cast<std::string>(o2::aod::femtodreamparticle::ParticleTypeName[mParticleType]);
+      folderName += static_cast<std::string>(mFolderSuffix[mFolderSuffixType]);
+
+      /// Particle-type specific histograms
+      if constexpr (mParticleType == o2::aod::femtodreamparticle::ParticleType::kTrack) {
+        /// Track histograms
+        mHistogramRegistry->add((folderName + "/hDCAxy_Primary").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_Daughter").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_Material").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_NotPrimary").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_Fake").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_DaughterLambda").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+        mHistogramRegistry->add((folderName + "/hDCAxy_DaughterSigmaplus").c_str(), "; #it{p}_{T} (GeV/#it{c}); DCA_{xy} (cm)", kTH2F, {ptAxis, DCAxyAxis});
+      } else if constexpr (mParticleType == o2::aod::femtodreamparticle::ParticleType::kV0) {
+        /// V0 histograms
+      } else if constexpr (mParticleType == o2::aod::femtodreamparticle::ParticleType::kCascade) {
+        /// Cascade histograms
+      } else {
+        LOG(fatal) << "FemtoDreamParticleHisto: Histogramming for requested object not defined - quitting!";
+      }
+    }
+  }
```

Comment:
In here you are replicating code which is probably not needed and error prone when the functionality were enhanced in the future
See my comments on the data model and on `fillQAMC`